conn.go

```diff
@@ -0,0 +1,233 @@
+		return nil, err
+	}
+
+	return wrappedStmt{opts: c.opts, query: query, parent: parent}, nil
+}
+
+func (c wrappedConn) Close() error {
+	return c.parent.Close()
+}
+
+func (c wrappedConn) Begin() (driver.Tx, error) {
+	tx, err := c.parent.Begin()
+	if err != nil {
+		return nil, err
+	}
+
+	return wrappedTx{opts: c.opts, parent: tx}, nil
+}
+
+func (c wrappedConn) BeginTx(ctx context.Context, opts driver.TxOptions) (tx driver.Tx, err error) {
+	if !c.hasOpExcluded(OpSQLTxBegin) {
+		span := c.GetSpan(ctx).NewChild(OpSQLTxBegin)
+		span.SetLabel("component", "database/sql")
+		start := time.Now()
+		defer func() {
+			span.SetError(err)
+			span.Finish()
+			c.Log(ctx, OpSQLTxBegin, "err", err, "duration", time.Since(start))
+		}()
+	}
+
+	if connBeginTx, ok := c.parent.(driver.ConnBeginTx); ok {
+		tx, err = connBeginTx.BeginTx(ctx, opts)
+		if err != nil {
+			return nil, err
+		}
+
+		return wrappedTx{opts: c.opts, ctx: ctx, parent: tx}, nil
+	}
+
+	tx, err = c.parent.Begin()
+	if err != nil {
+		return nil, err
+	}
+
+	return wrappedTx{opts: c.opts, ctx: ctx, parent: tx}, nil
+}
+
+func (c wrappedConn) PrepareContext(ctx context.Context, query string) (stmt driver.Stmt, err error) {
+	if !c.hasOpExcluded(OpSQLPrepare) {
+		span := c.GetSpan(ctx).NewChild(OpSQLPrepare)
+		span.SetLabel("component", "database/sql")
+		start := time.Now()
+		defer func() {
+			span.SetError(err)
+			span.Finish()
+			logQuery(ctx, c.opts, OpSQLPrepare, query, err, nil, start)
+		}()
+	}
+
+	if connPrepareCtx, ok := c.parent.(driver.ConnPrepareContext); ok {
+		stmt, err := connPrepareCtx.PrepareContext(ctx, query)
+		if err != nil {
+			return nil, err
+		}
+
+		return wrappedStmt{opts: c.opts, ctx: ctx, query: query, parent: stmt}, nil
+	}
+
+	return c.Prepare(query)
+}
+
+func (c wrappedConn) Exec(query string, args []driver.Value) (driver.Result, error) {
+	if execer, ok := c.parent.(driver.Execer); ok {
+		res, err := execer.Exec(query, args)
+		if err != nil {
+			return nil, err
+		}
+
+		return wrappedResult{opts: c.opts, parent: res}, nil
+	}
+
+	return nil, driver.ErrSkip
+}
+
+func (c wrappedConn) ExecContext(ctx context.Context, query string, args []driver.NamedValue) (r driver.Result, err error) {
+	if !c.hasOpExcluded(OpSQLConnExec) {
+		span := c.GetSpan(ctx).NewChild(OpSQLConnExec)
+		span.SetLabel("component", "database/sql")
+		span.SetLabel("query", query)
+		if !c.OmitArgs {
+			span.SetLabel("args", formatArgs(args))
+		}
+		start := time.Now()
+		defer func() {
+			span.SetError(err)
+			span.Finish()
+
+			logQuery(ctx, c.opts, OpSQLConnExec, query, err, args, start)
+		}()
+	}
+
+	if execContext, ok := c.parent.(driver.ExecerContext); ok {
+		res, err := execContext.ExecContext(ctx, query, args)
+		if err != nil {
+			return nil, err
+		}
+
+		return wrappedResult{opts: c.opts, ctx: ctx, parent: res}, nil
+	}
+
+	// Fallback implementation
+	dargs, err := namedValueToValue(args)
+	if err != nil {
+		return nil, err
+	}
+
+	select {
```

Comment:
If the goal of your select is to see whether `ctx` is already done and proceed with `Exec` if not wouldn't it be more clear to the reader if it was written as:
```go
select {
  case <-ctx.Done():
    return nil, ctx.Err()
  default:
    return c.Exec(query, dargs)
}
//... no more code in ExecContext
```

Right now that select might be misunderstood as if it does `return nil, ctx.Err()` for both cases if one would assume that there is a similar flow as in `switch` (it isn't IIRC).

If would prefer having the `return` statement placed outside of that `select` I'd suggest moving the `default` to the last position to avoid any confusion:
```go
select {
  case <-ctx.Done():
    return nil, ctx.Err()
  default:
}

return c.Exec(query, dargs)
```